### PR TITLE
chore: add macos daily build

### PR DESCRIPTION
**Pull request by romange**
It compiles most of the code though some linking problems still exist.

### CODE REVIEW

File: patches/lua-v5.4.4.patch [hunk 1]
```diff
@@ -15,13 +15,15 @@ diff --git a/makefile b/makefile
...
 +++ b/makefile
-@@ -66,13 +66,25 @@ LOCAL = $(TESTS) $(CWARNS)
+@@ -66,13 +66,26 @@ LOCAL = $(TESTS) $(CWARNS)
 
 
  # enable Linux goodies
 -MYCFLAGS= $(LOCAL) -std=c99 -DLUA_USE_LINUX -DLUA_USE_READLINE
 +MYCFLAGS= $(LOCAL) -std=c99 -g -O2 -DLUA_USE_LINUX
- MYLDFLAGS= $(LOCAL) -Wl,-E
+-MYLDFLAGS= $(LOCAL) -Wl,-E
```
Comment (romange) [inline patches/lua-v5.4.4.patch:24]: this is the only change here

File: src/core/search/indices.cc [hunk 2]
```diff
@@ -74,7 +74,8 @@ absl::flat_hash_set<std::string> ICUTokenizeWords(std::string_view text) {
 
 // Convert string to lowercase with ICU library
 std::string ICUToLowercase(string_view input) {
-  icu::UnicodeString uStr = icu::UnicodeString::fromUTF8(input);
+  icu::UnicodeString uStr =
+      icu::UnicodeString::fromUTF8(icu::StringPiece(input.data(), input.size()));
```
Comment (romange) [inline src/core/search/indices.cc:78]: older icu lib can not convert from std::string_view

File: src/server/CMakeLists.txt [hunk 3]
```diff
@@ -21,10 +21,16 @@ add_library(dfly_transaction db_slice.cc malloc_stats.cc engine_shard_set.cc blo
             )
 cxx_link(dfly_transaction dfly_core strings_lib)
 
+
+if (NOT APPLE)
```
Comment (romange) [inline src/server/CMakeLists.txt:25]: remove search functionality from MacOS for now.

File: src/core/CMakeLists.txt [hunk 4]
```diff
@@ -24,4 +24,7 @@ cxx_test(sorted_map_test dfly_core LABELS DFLY)
 cxx_test(bptree_set_test dfly_core LABELS DFLY)
 cxx_test(score_map_test dfly_core LABELS DFLY)
 
-add_subdirectory(search)
+# We have some linking problems with search on Apple
+if (NOT APPLE)
+  add_subdirectory(search)
+endif()
```
Comment (dranikpg) [inline src/core/CMakeLists.txt:30]: Reminds me of the fact that our target is still called `query_parser` 😅

File: src/core/search/indices.cc (hunk 2 above)
Comment (dranikpg) [reply src/core/search/indices.cc:78]: As long as search is excluded we can actually exclude the icu dependency as well

**Review by dranikpg — DISMISSED** [review]

**Review by dranikpg — APPROVED** [review]

File: src/server/rdb_save.cc [hunk 5]
```diff
@@ -1022,7 +1024,7 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
   // we can not exit on io-error since we spawn fibers that push data.
   // TODO: we may signal them to stop processing and exit asap in case of the error.
 
-  while (record = records_popper.Pop()) {
+  while ((record = records_popper.Pop())) {
```
Comment (chakaz) [inline src/server/rdb_save.cc:1027]: You could also do:

```cpp
while (record = records_popper.Pop(); record.has_value()) {
```

or

```cpp
while (record = records_popper.Pop(); record) {
```

Your call, but I think the double parenthesis can be confusing

File: src/server/rdb_save.cc [hunk 6]
```diff
@@ -1037,7 +1039,7 @@ error_code RdbSaver::Impl::ConsumeChannel(const Cancellation* cll) {
       if (io_error) {
         break;
       }
-    } while (record = records_popper.TryPop());
+    } while ((record = records_popper.TryPop()));
```
Comment (chakaz) [inline src/server/rdb_save.cc:1042]: Ditto

**Review by chakaz — APPROVED** [review]

File: src/server/rdb_save.cc (hunk 5 above)
Comment (romange) [reply src/server/rdb_save.cc:1027]: I just fixed warnings by compiler and did what he told me to :)

**Pull request merged**